### ISSUE TITLE
create a branch directly from commit on the create branch API

### ISSUE THREAD

**Issue by viletyy** [go-gitea/gitea#22139]
### Feature Description

While working with gitea, we discovered that the interface does not support creating a branch directly from a commit. You can do this using the git command line, but not every user uses the command line.

**Comment by CodeDoctorDE** [comment]
![grafik](https://user-images.githubusercontent.com/20452814/210727298-a930410f-c32c-499b-8970-37a620cc89f0.png)
You mean like this? You can do this by clicking on a commit and click on Operations

**Comment by viletyy** [comment]
> ![grafik](https://user-images.githubusercontent.com/20452814/210727298-a930410f-c32c-499b-8970-37a620cc89f0.png) You mean like this? You can do this by clicking on a commit and click on Operations
yes, just like this, but we want to be able to do this through a API

**Comment by viletyy** [comment]
> ![grafik](https://user-images.githubusercontent.com/20452814/210727298-a930410f-c32c-499b-8970-37a620cc89f0.png) You mean like this? You can do this by clicking on a commit and click on Operations

yes, just like this, but we want to be able to do this through a API

**Comment by ares1977** [comment]
It seems it is only implemented in the web, not in the API.

web:

https://github.com/go-gitea/gitea/blob/f8c1e14a136c9702f459a36a1ea2e75fedd123af/routers/web/repo/branch.go#L361-L372

API:

https://github.com/go-gitea/gitea/blob/f8c1e14a136c9702f459a36a1ea2e75fedd123af/routers/api/v1/repo/branch.go#L179-L181

**Comment by Mik4sa** [comment]
I'm also interested in this one. This there any state for it?

**Comment by viletyy** [comment]
> I'm also interested in this one. This there any state for it?

I'm very sorry that there is no progress at this one.
